sale_coupon_advanced_delivery/models/sale_order.py

```diff
@@ -0,0 +1,25 @@
+# Copyright 2020 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import models
+
+
+class SaleOrder(models.Model):
+    """Extend to make promotions recompute aware of delivery lines."""
+
+    _inherit = "sale.order"
+
+    def get_update_pricelist_order_lines(self):
+        """Extend to ignore delivery lines."""
+        order_lines = super().get_update_pricelist_order_lines()
+        return order_lines.filtered(lambda r: not r.is_delivery)
+
+    def recompute_coupon_lines(self):
+        """Extend to trigger delivery price update."""
+        super().recompute_coupon_lines()
+        ChooseDeliveryCarrier = self.env["choose.delivery.carrier"]
```

Comment:
```suggestion
        choose_delivery_carrier = self.env["choose.delivery.carrier"]
```
can you please change this camel case to something more usual in Odoo's world?

Reply:
IMO initial naming is fine, it becomes more usable in odoo
if you update to the recommended add also "_obj"